src/ota.c

```diff
@@ -88,10 +96,49 @@ const struct golioth_ota_component *golioth_ota_find_component(
     return found;
 }
 
+static void ota_manifest_timer_expiry(golioth_sys_timer_t timer, void *user_arg)
+{
+    struct manifest_get_args *get_args = user_arg;
+
+    enum golioth_status status = golioth_ota_get_manifest_async(get_args->client,
+                                                                get_args->cb,
+                                                                get_args->arg);
+
+    if (GOLIOTH_OK != status)
+    {
+        GLTH_LOGE(TAG, "Periodic OTA manifest fetch failed");
+    }
+
+    golioth_sys_timer_reset(timer);
+}
+
 enum golioth_status golioth_ota_observe_manifest_async(struct golioth_client *client,
                                                        golioth_get_cb_fn callback,
                                                        void *arg)
 {
+    manifest_timer_arg.client = client;
+    manifest_timer_arg.cb = callback;
+    manifest_timer_arg.arg = arg;
+
+    if (!manifest_timer_initialized && CONFIG_GOLIOTH_OTA_MANIFEST_OBSERVE_PERIOD_S > 0)
+    {
+        struct golioth_timer_config cfg =
+        {
+            .name = "ota_manifest_timer",
+            .expiration_ms = CONFIG_GOLIOTH_OTA_MANIFEST_OBSERVE_PERIOD_S * 1000,
+            .fn = ota_manifest_timer_expiry,
+            .user_arg = &manifest_timer_arg,
+        };
+        manifest_timer = golioth_sys_timer_create(&cfg);
```

Comment:
My first reaction was that it is a bit out of place for this timer / polling logic to be handled in the OTA client methods and coupled to establishing an observation. However, in thinking about it more, if we elevate the concept of "observing" beyond the CoAP implementation of observations and instead make observe mean the discussed "polling + short circuit" behavior then I think managing the timer within the OTA client makes sense 👍🏻